src/main/java/net/javadiscord/javabot/systems/notification/NotificationService.java

```diff
@@ -1,35 +1,59 @@
 package net.javadiscord.javabot.systems.notification;
 
 import lombok.extern.slf4j.Slf4j;
+import net.dv8tion.jda.api.entities.Guild;
 import net.dv8tion.jda.api.entities.MessageChannel;
-import net.dv8tion.jda.api.entities.MessageEmbed;
 import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.requests.restaction.MessageAction;
+import net.javadiscord.javabot.util.ExceptionLogger;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
 
+import java.util.function.Function;
 
 /**
- * Abstract class used for sending Notifications to Discord Users and Channels.
+ * Handles all types of guild & user notifications.
  */
-@Slf4j
-public abstract class NotificationService {
-	void sendDirectMessageNotification(User user, MessageEmbed message) {
-		user.openPrivateChannel().queue(
-				channel -> sendMessageChannelNotification(channel, message),
-				error -> log.warn("Could not send private Notification to User " + user.getAsTag())
-		);
+public class NotificationService {
```

Comment:
Make this class `final` to document the fact that is is not possible to create subclasses of `NotificationService` (due to the private constructor)